app.rb

```diff
@@ -0,0 +1,172 @@
+require './book'
+require './student'
+require './teacher'
+require './rental'
+require './person'
+
+class App
+  attr_reader :people, :books
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def show_menu
+    puts 'Welcome to school library app!'
+    puts ''
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person ID'
+    puts '7 - Exit'
+    @number = gets
+    until @number.to_i.between?(1, 7)
+      puts 'Please write a number between 1 and 7'
+      @number = gets
+    end
+  end
+
+  def options
+    case @number.to_i
+    when 1
+      list_all_books
+      run
+    when 2
+      list_all_people
+      run
+    when 3
+      create_person
+      run
+    when 4
+      create_book
+      run
+    when 5
+      create_rental
+      run
+    when 6
+      list_all_rentals_for_id
+      run
+    end
+  end
+
+  def run
+    show_menu
+    if @number.to_i == 7
+      exit
+    else
+      options
+    end
+  end
+
+  def create_person
+    print 'Please choose 1 for create a student or 2 for create a teacher: '
+    teacher_or_student = gets.chomp.to_i
+    case teacher_or_student
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid option, try again'
+      create_person
+    end
+  end
+
+  def create_student
+    print 'Age?: '
+    age = gets.chomp.to_i
+    print 'Name?: '
+    name = gets.chomp
+    print 'Has parent permission?[Y/N]: '
+    parent_permission = gets.chomp
+    case parent_permission
+    when 'y' || 'Y'
+      new_student = Student.new(age, name, true)
+      @people << new_student
+      puts 'Person created sucessfully'
+    when 'n' || 'N'
+      new_student = Student.new(age, name, false)
+      @people << new_student
+      puts 'Person created sucessfully'
+    else
+      puts 'Invalid, please try again'
+      create_student
+    end
+  end
+
+  def create_teacher
+    print 'Age?: '
+    age = gets.chomp.to_i
+    print 'Name?: '
+    name = gets.chomp
+    print 'Specialization?: '
+    specialization = gets.chomp
+    new_teacher = Teacher.new(age, specialization, name)
+    @people << new_teacher
+    puts 'Person created sucessfully'
```

Comment:
- [ ] After creating a person student and teacher it didn't return the successful message. Please 🙏🏽 fix this issue for a better UX.